proc-macros/src/attributes.rs

```diff
@@ -40,6 +40,12 @@ pub(crate) struct Argument {
 	pub tokens: TokenStream2,
 }
 
+#[derive(Debug, Clone)]
+pub enum ParamFormat {
```

Comment:
This is a total nitpick, but what do you think about naming this `ParameterKind`? It's nor really a "format" but more akin to a type, or mode of operation of sorts.
Also: docs missing here.

Reply:
I called it format because I was thinking of it in terms of parameter layout and not necessarily as types, but I've absolutely no hard feelings about this. I've renamed it to `ParamKind`